compiler/rustc_target/src/spec/apple_base.rs

```diff
@@ -179,20 +179,66 @@ pub fn opts(os: &'static str, arch: Arch) -> TargetOptions {
     }
 }
 
-pub fn deployment_target(target: &Target) -> Option<String> {
+pub fn sdk_version(platform: u32) -> Option<(u32, u32)> {
+    match platform {
+        object::macho::PLATFORM_MACOS => Some((13, 1)),
+        object::macho::PLATFORM_IOS
+        | object::macho::PLATFORM_IOSSIMULATOR
+        | object::macho::PLATFORM_TVOS
+        | object::macho::PLATFORM_TVOSSIMULATOR
+        | object::macho::PLATFORM_MACCATALYST => Some((16, 2)),
+        object::macho::PLATFORM_WATCHOS | object::macho::PLATFORM_WATCHOSSIMULATOR => Some((9, 1)),
+        _ => None,
+    }
+}
```

Comment:
Another numbers question, what selected these? LLVM appears to dynamically read them from the XCode SDK when running on macOS, or just put nothing in which is what I would have expected here too. These ones are going to go stale fast, for example XCode 14.3 has 13.3 and `clang` embeds that.

Reply:
I pulled these from the same version of Xcode that was previously used to hardcode the original catalyst value here https://github.com/rust-lang/rust/pull/114114/files#diff-6bbf6edd930f26d3b180eb5c250b0e48d8f3c5eb474e3274909ef6ae6f0d1e61L355 

I'm not sure what we should do as far as reading these dynamically. Folks could build a rust library targeting iOS without Xcode installed, or from a different OS, in which case I suppose we could fall back to something like this as a default.

The only place I know that Apple uses this metadata is frameworks like UIKit that ships with iOS has "linked against SDK" checks, where they potentially change behavior if you've rebuilt your apps with the newest SDK. 

In this case of this change, specifically for the metadata objects, this version likely won't end up in the final binary since the final `ld -platform_version ios X Y` argument will take precedence (unless you build a dylib for iOS, but even then the rmeta files will likely be dropped during linking?)

Reply:
> The only place I know that Apple uses this metadata is frameworks like UIKit that ships with iOS has "linked against SDK" checks, where they potentially change behavior if you've rebuilt your apps with the newest SDK.
> 
>In this case of this change, specifically for the metadata objects, this version likely won't end up in the final binary since the final ld -platform_version ios X Y argument will take precedence...

Hm am I understanding right that these don't make it past the linker then for libraries? That makes it "fine enough" to just hardcode these to something `ld` won't complain about. Maybe at some point we could just read XCode's bundle version (when available) and match some sane defaults based on it since we only need "broadly compatible" with the current toolchain.

> (unless you build a dylib for iOS, but even then the rmeta files will likely be dropped during linking?)

I just compiled a dylib for aarch64 iOS using 1.71 and the `sdk` part of `LC_BUILD_VERSION` looked fine. Appears to just be doing the same thing where `ld` is putting in the right versions it has knowledge of so I imagine there's nothing to cause concern.

Reply:
> Hm am I understanding right that these don't make it past the linker then for libraries?

If it's a static lib and you incorporate that into your final link of a standard iOS app the final `-platform_version` argument will definitely take over this one. I don't even think in that case these values will be read.

> Maybe at some point we could just read Xcode's bundle version (when available) and match some sane defaults based on it since we only need "broadly compatible" with the current toolchain.

Yea I think that would work fine it just feels like a lot of work if there isn't a noticeable win

> Appears to just be doing the same thing where `ld` is putting in the right versions it has knowledge of so I imagine there's nothing to cause concern.

After you said this I did a quick test as well and it looks like that codepath with cargo ends up calling clang to link, which does the discovery from Xcode, nice!

Reply:
> Yea I think that would work fine it just feels like a lot of work if there isn't a noticeable win

Yeah, very low priority IMO. I'm mostly interested in the idea just to prevent this file from bitrotting into a bunch of dead constants or some future split in Apple's XCode toolchains. Either way, nothing to do about it here.

Reply:
Actually, maybe just add a comment saying these won't affect people's builds and just need to be good enough?

Reply:
Done!